src/DeferredFutures.jl

```diff
@@ -1,35 +1,88 @@
 module DeferredFutures
 
-export DeferredFuture
+export @defer, DeferredChannel, DeferredFuture
 
 using AutoHashEquals
 
-@auto_hash_equals immutable DeferredFuture <: Base.AbstractRemoteRef
+abstract DeferredRemoteRef <: Base.AbstractRemoteRef
+
+@auto_hash_equals immutable DeferredFuture <: DeferredRemoteRef
+    outer::Future
+end
+
+DeferredFuture(pid::Integer=myid()) = DeferredFuture(Future(pid))
+
+@auto_hash_equals immutable DeferredChannel <: DeferredRemoteRef
     outer::Future
+    func::Function      # A function the generates the RemoteChannel
+end
+
+function DeferredChannel(f::Function, pid::Integer=myid())
```

Comment:
DeferredFutures are API-compatible with Futures. To be API-compatible with RemoteChannels, the function accepted should be the one which creates the inner channel, not the one which creates the RemoteChannel.